libs/elasticsearch/README.md

```diff
@@ -110,3 +110,71 @@ chat_history = ElasticsearchChatMessageHistory(
     es_api_key="your-api-key",
 )
 ```
+
+
+### ElasticsearchCache
+
+A caching layer for LLMs that exploits Elasticsearch.
+
+Simple example:
+
+```python
+from langchain.globals import set_llm_cache
+from langchain_elasticsearch.cache import ElasticsearchCache
+from elasticsearch import Elasticsearch
+
+es_client = Elasticsearch(hosts="http://localhost:9200")
+set_llm_cache(
+    ElasticsearchCache(
+        es_connection=es_client, 
+        es_index="llm-chat-cache", 
+        metadata={"project": "my_chatgpt_project"}
+    )
+)
+```
+
+The `es_index` parameter can also take aliases. This allows to use the 
+[ILM: Manage the index lifecycle](https://www.elastic.co/guide/en/elasticsearch/reference/current/index-lifecycle-management.html)
+that we suggest to consider for managing retention and controlling cache growth.
+
+Look at the class docstring for all parameters.
+
+#### Index the generated text
+
+The cached data won't be searchable by default.
+The developer can customize the building of the Elasticsearch document in order to add indexed text fields,
+where to put, for example, the text generated by the LLM.
+
+This can be done by subclassing end overriding methods.
+The new cache class can be applied also to a pre-existing cache index:
+
+```python
+from langchain_elasticsearch.cache import ElasticsearchCache
+from elasticsearch import Elasticsearch
+from langchain_core.caches import RETURN_VAL_TYPE
+from typing import Any, Dict, List
+from langchain.globals import set_llm_cache
+import json
```

Comment:
Could you [isort](https://pycqa.github.io/isort/) these imports?

Reply:
done with ruff